tests/serac_boundary_cond.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+#include "common/common.hpp"
+#include "mfem.hpp"
+
+namespace serac {
+
+TEST(boundary_cond, simple_repeated_dofs)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+  constexpr int      N    = 15;
+  constexpr int      ATTR = 1;
+  mfem::Mesh         mesh(N, N, mfem::Element::TRIANGLE);
+  mfem::ParMesh      par_mesh(MPI_COMM_WORLD, mesh);
+  FiniteElementState state(par_mesh);
+
+  for (int i = 0; i < par_mesh.GetNE(); i++) {
+    par_mesh.SetAttribute(i, ATTR);
```

Comment:
I believe this changes the attribute on the volumetric element - not the boundary. I think you want something like this
```
for (int i = 0; i < num_boundary_elements; i++) {
  boundary[i]->SetAttribute(ATTR);
}
````
(only using appropriate accessor functions)